src/olympia/files/utils.py

```diff
@@ -1043,7 +1043,13 @@ def parse_addon(
             msg = gettext(
                 'The type (%s) does not match the type of your add-on on AMO (%s)'
             )
-            raise forms.ValidationError(msg % (parsed['type'], addon.type))
```

Comment:
This is a drive-by fix for this fairly obscure/rare error message which used to return the constant instead of the "pretty" name. See corresponding test fix. 

Note with 2FA enforcement enabled, it can be difficult to trigger anyway: if your user doesn't have 2FA, and you're trying to upload an extension as a version of an existing theme, you'd get the `Please add two-factor authentication to your account to submit extensions` error message instead, as devhub would detect you're using a theme specific flow to upload an extension. It's still relevant if doing the opposite (uploading a theme as a version of an extension) or when using other types though.